keps/sig-cluster-lifecycle/wgs/1815-componentconfig-objectmeta/README.md

```diff
@@ -0,0 +1,694 @@
+
+Related issues for context:
+
+- https://github.com/kubernetes/kubeadm/issues/1727
+
+### Goals
+
+<!--
+List the specific goals of the KEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+- Add a standard set of ObjectMeta fields (name) to the ComponentConfig API's.
+  - kubelet
+  - kube-proxy
+  - kube-scheduler
+  - kube-controller-manager
+  - kube-apiserver
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Graduate the API versions
+  - We will implement ObjectMeta within the current version of each ComponetConfig API
+- Implement every possible ObjectMeta field for ComponentConfig API's
+  - As these configs are singular per component, per cluster, they won't have or need the full suite of ObjectMeta similar to core Kubernetes API's (ie. Pods, Servcies, etc.)
+
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+- for component in [kubelet kubeproxy kubecontrollermanager kubeapiserver kubescheduler]
+  - Add the following ObjectMeta fields
+    - name
+
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+None. The ObjectMeta schema is standardized throughout the Kubernetes codebase and is already compatible with approaches codified by Kustomize.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Add the ObjectMeta definition to the component-base repo to be consumed by all component config API's.
+
+k8s.io/component-base/config/v1alpha1/types.go
+
+```go
+// ObjectMeta is metadata that all persisted resources must have, which includes all objects
+// users must create. This is a copy of customizable fields from metav1.ObjectMeta.
+type ObjectMeta struct {
+  // Name must be unique within a component instance. Name is primarily 
```

Comment:
nit: use consistent indentation of comments and code here and elsewhere

Reply:
What was the inconsistency here? `ObjectMeta` comments match to the definition of the `ObjectMeta` struct, as do the `Name` comments match to the `Name` field within the struct (indented to match the indentation of the field)

Reply:
Sorry for any confusion. Most of these lines are indented two spaces, but starting with `// More info` they are indented one tab character.